### DISCUSSION TITLE
How to use git dirty status to determine when to run a task?

### DISCUSSION THREAD

**Discussion by ssbarnea** [go-task/task#741] [Q&A]
Using file-glob patterns to determine when to rebuild (rerun task) is far from easy but on the other hand using git ability to detect changes is a very powerful feature, as users already have `.gitignore` exclude patterns added and the tool is more likely to be able to quickly detect a state change.

Is there a way to use git to determine these? Does anyone have some practical examples?

**Comment by priyamsahoo** [comment]
That is a great explanation of the feature that we wanted in our project.
To explain it furthermore, we want to run the 'lint' job only when we change files other than the ones mentioned in '.gitignore' rather than running it always.

**Comment by mgbowman** [comment]
I have done something like this before...

```
  create-pr:
    status:
      # only run when adding (A) or modifying (M) generated files in OUTPUT_DIR
      - '! git status -s | egrep -q "[AM][ ]+{{.OUTPUT_DIR}}"'
```
